Activate native code support for OpenBSD/aarch64. Can be removed when updating
to ocaml 4.14.

Backport of:
https://github.com/ocaml/ocaml/commit/c288f707e0c0722172191fc717977f3db85f131b

Index: configure.ac
--- configure.ac.orig
+++ configure.ac
@@ -1020,6 +1020,7 @@ AS_IF([test x"$shared_libraries_supported" = 'xtrue'],
     [earm*-*-netbsd*], [natdynlink=true],
     [aarch64-*-linux*], [natdynlink=true],
     [aarch64-*-freebsd*], [natdynlink=true],
+    [aarch64-*-openbsd*], [natdynlink=true],
     [riscv*-*-linux*], [natdynlink=true])])
 
 # Try to work around the Skylake/Kaby Lake processor bug.
@@ -1122,6 +1123,8 @@ AS_CASE([$host],
     [arch=arm64; system=linux],
   [aarch64-*-freebsd*],
     [arch=arm64; system=freebsd],
+  [aarch64-*-openbsd*],
+    [arch=arm64; system=openbsd],
   [x86_64-*-cygwin*],
     [arch=amd64; system=cygwin],
   [riscv64-*-linux*],
